### PR TITLE
Improve graphql service BBEs descriptions

### PR DESCRIPTION
**Pull request by MohamedSabthar**
## Purpose
Related to https://github.com/ballerina-platform/ballerina-standard-library/issues/3798

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Review by ThisaruGuruge — APPROVED** [review]

File: examples/graphql-interfaces/graphql_interfaces.md [hunk 1]
```diff
@@ -1,8 +1,6 @@
...
-
-This example shows how to define an interface `Profile` and then implement the `Teacher` and `Student` classes using that interface.
+The Ballerina GraphQL module allows defining GraphQL interface types. An interface specifies a set of fields that multiple object types can include. In Ballerina, interfaces are defined using `distinct` `service` objects, and the fields of the interfaces are defined as resource method definitions. Objects that are implementing the interfaces must implement the `resource` methods defined in the `service` objects. The Ballerina type inclusion includes the interface type to an object type. Use an interface to return a type that consists of a set of possible types with common fields.
```
Comment (praneesha) [inline examples/graphql-interfaces/graphql_interfaces.md:3]:
```suggestion
The Ballerina GraphQL module allows defining GraphQL interface types. An interface specifies a set of fields that multiple object types can include. In Ballerina, interfaces are defined using `distinct` `service` objects and the fields of the interfaces are defined as resource method definitions. Objects that are implementing the interfaces must implement the `resource` methods defined in the `service` objects. The Ballerina type inclusion includes the interface type to an object type. Use an interface to return a type that consists of a set of possible types with common fields.
```
This paragraph is too lengthy. Better to break it into two.

File: examples/graphql-mutations/graphql_mutations.bal [hunk 2]
```diff
@@ -19,7 +19,7 @@ service /graphql on new graphql:Listener(9090) {
         return profiles.get(id);
     }
 
-    // A remote method represents a field in the root `Mutation` operation. This remote method will
+    // A `remote` method represents a field in the root `Mutation` operation. This `remote` method will
     // update the name for the given profile ID, and returns the updated `Profile` value. If the ID
```
Comment (praneesha) [inline examples/graphql-mutations/graphql_mutations.bal:23]:
```suggestion
    // update the name for the given profile ID and returns the updated `Profile` value. If the ID
```

File: examples/graphql-returning-record-values/graphql_returning_record_values.md [hunk 3]
```diff
@@ -1,10 +1,7 @@
 # GraphQL service - Record as output object
 
-In a Ballerina GraphQL service, a record type can be used to represent an `OBJECT` type in a GraphQL schema. The type name and the field names are mapped one-to-one from the Ballerina to GraphQL. Types are generated by analyzing the types that are used in a Ballerina GraphQL service recursively. Use a record type to represent an `OBJECT` type only when all fields of that object type do not have any input arguments or the field resolution does not require any complex logic execution.
+The Ballerina GraphQL module allows returning `record` types from `resource` or `remote` methods of `graphql:Service`. These `record` types are mapped to GraphQL output object types in the GraphQL schema, where the type name and the field names are mapped one-to-one from the Ballerina to GraphQL. Use a `record` type to represent a GraphQL output object type only when all fields of that object type do not have any input arguments or the field resolution does not require any complex logic execution. `record` type is preferred over the `service` object type, in this case, as it makes the code more concise.
```
Comment (praneesha) [inline examples/graphql-returning-record-values/graphql_returning_record_values.md:3]:
```suggestion
The Ballerina GraphQL module allows returning `record` types from the `resource` or `remote` methods of the `graphql:Service`. These `record` types are mapped to GraphQL output object types in the GraphQL schema in which the type name and the field names are mapped one-to-one from Ballerina to GraphQL. Use a `record` type to represent a GraphQL output object type only when all fields of that object type do not have any input arguments or the field resolution does not require any complex logic execution. The `record` type is preferred over the `service` object type in this case as it makes the code more concise.
```

File: examples/graphql-returning-service-objects/graphql_returning_service_objects.md [hunk 4]
```diff
@@ -1,10 +1,8 @@
...
+The Ballerina GraphQL module allows returning `service` objects from `resource` or `remote` methods of `graphql:Service`. These `service` objects are mapped to GraphQL output object types in the GraphQL schema. Each `resource` method in the returned `service` object becomes a field in the created GraphQL output object type. The `resource` methods in these `service` types can have input parameters. These input parameters are mapped to arguments in the corresponding field. Use a `service` type to represent a GraphQL output object type when a field of that GraphQL output object type has any input arguments. Further, using a `service` type to represent a GraphQL output object type offers the flexibility of organizing complex logic.
```
Comment (praneesha) [inline examples/graphql-returning-service-objects/graphql_returning_service_objects.md:3]:
```suggestion
The Ballerina GraphQL module allows returning `service` objects from the `resource` or `remote` methods of the `graphql:Service`. These `service` objects are mapped to GraphQL output object types in the GraphQL schema. Each `resource` method in the returned `service` object becomes a field in the created GraphQL output object type. The `resource` methods in these `service` types can have input parameters. These input parameters are mapped to arguments in the corresponding field. Use a `service` type to represent a GraphQL output object type when a field of that GraphQL output object type has any input arguments. Further, using a `service` type to represent a GraphQL output object type offers the flexibility of organizing complex logic.
```
This paragraph is too lengthy. Better to break it into two.

File: examples/graphql-subscriptions/graphql_subscriptions.md [hunk 5]
```diff
@@ -1,8 +1,7 @@
...
+The Ballerina GraphQL module allows defining GraphQL `Subscription` operations. A resource method with the `subscribe` accessor inside a GraphQL service represents a field in the root `Subscription` type. Therefore, If a resource method with the `subscribe` accessor is present inside the Ballerina GraphQL service, the auto-generated schema will have a `Subscription` type. Each resource method with a `subscribe` accessor in the service is added as a field of the `Subscription` type. The field name will be the resource method name and the field type will be the constraint type of the `stream` returned from the resource method. Not returning a `stream` type from a resource method having a `subscribe` accessor results in a compilation error. Use a subscription operation to monitor small, incremental changes to large objects or to obtain low-latency, real-time updates.
```
Comment (praneesha) [inline examples/graphql-subscriptions/graphql_subscriptions.md:3]:
```suggestion
The Ballerina GraphQL module allows defining GraphQL `Subscription` operations. A resource method with the `subscribe` accessor inside a GraphQL service represents a field in the root `Subscription` type. Therefore, If a resource method with the `subscribe` accessor is present inside the Ballerina GraphQL service, the auto-generated schema will have a `Subscription` type. Each resource method with a `subscribe` accessor in the service is added as a field of the `Subscription` type. The field name will be the resource method name and the field type will be the constraint type of the `stream` returned from the resource method. Not returning a `stream` type from a resource method having a `subscribe` accessor results in a compilation error. Use a subscription operation to monitor small, incremental changes to large objects or to obtain low-latency, real-time updates.
```
This paragraph is too lengthy. Better to break it into two.

File: examples/graphql-interfaces/graphql_interfaces.md (hunk 1 above)
Comment (MohamedSabthar) [reply examples/graphql-interfaces/graphql_interfaces.md:3]: @praneesha Is there a limitation on how many words should be in BBE description paragraph ?

File: examples/graphql-mutations/graphql_mutations.bal (hunk 2 above)
Comment (MohamedSabthar) [reply examples/graphql-mutations/graphql_mutations.bal]: Updated

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**